app/src/components/tag/TagList.tsx

```diff
@@ -0,0 +1,21 @@
+import React, { forwardRef, Ref } from "react";
+import { TagList as AriaTagList, TagListProps } from "react-aria-components";
+import { css } from "@emotion/react";
+
+const tagListCSS = css`
+  display: flex;
+  align-items: center;
+  flex-wrap: wrap;
+  gap: var(--ac-global-dimension-size-50);
+  height: 28px;
+`;
+function TagList<T extends object>(
```

Comment:
I think this will fix inference

```suggestion
function TagList<T extends Record<string, unknown>>(
```

Reply:
The base props extends object. you actually need to pass it through the ref forwarding:

```typescript
// Use forwardRef with the generic type to ensure the type propagates
const _TagList = forwardRef(TagList) as <T extends object>(
  props: TagListProps<T> & { ref?: Ref<HTMLDivElement> }
) => JSX.Element;
```